modules/gapi/src/streaming/onevpl/data_provider_dispatcher.cpp

```diff
@@ -0,0 +1,68 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#ifdef HAVE_ONEVPL
+
+#include "streaming/onevpl/data_provider_dispatcher.hpp"
+#include "streaming/onevpl/file_data_provider.hpp"
+#include "streaming/onevpl/demux/async_mfp_demux_data_provider.hpp"
+#include "logger.hpp"
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+IDataProvider::Ptr DataProviderDispatcher::create(const std::string& file_path,
+                                                  const std::vector<CfgParam> &cfg_params) {
+    GAPI_LOG_INFO(nullptr, "try select suitable IDataProvider for source: " <<
+                           file_path);
+
+    IDataProvider::Ptr provider;
+
+    // Look-up CodecId from input params
+    // If set then raw data provider is preferred
+    GAPI_LOG_DEBUG(nullptr, "try find explicit cfg param\"mfxImplDescription.mfxDecoderDescription.decoder.CodecID\"");
```

Comment:
should this sting be defined only once somewhere?

Reply:
will add in similar way as  `queue_capacity` in future PRs
https://github.com/opencv/opencv/pull/21049/files#diff-3b97bff710e57efd4529da47be268daad0dddeeda82225e6354c18b93808b786R63